### DISCUSSION TITLE
Sveltekit and components - SvelteKitError: Not found: /__cypress/src/index.html

### DISCUSSION THREAD

**Discussion by magick93** [cypress-io/cypress#29282] [Questions and Help]
Hello

I'm trying to get setup testing components with Sveltekit. 

However in both headless and browser testing I'm getting the error `SvelteKitError: Not found: /__cypress/src/index.html`. 

I do have `cypress/support/component-index.html` with the following contents:

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width,initial-scale=1.0">
    <title>Components App</title>
  </head>
  <body>
    <div data-cy-root></div>
  </body>
</html>
```

And I have the following config settings:

```
component: {
    devServer: {
      framework: "svelte",
      bundler: "vite",
    },
    indexHtmlFile: 'cypress/support/component-index.html',
    supportFile: 'cypress/cypress/support/component.ts',
  },
  ```

Any help is appreciated

**Comment by portyu9** [comment]
Hi @magick93 

The `component-index.html` itself looks fine. The main issue is that Cypress is picking up the **SvelteKit Vite configuration**, so SvelteKit tries to route Cypress's internal `/__cypress/src/index.html` URL and returns the `SvelteKitError`.

Give Cypress a plain Svelte Vite config instead:

```ts
import { defineConfig } from 'cypress'
import { svelte } from '@sveltejs/vite-plugin-svelte'

export default defineConfig({
  component: {
    devServer: {
      framework: 'svelte',
      bundler: 'vite',
      viteConfig: {
        plugins: [svelte()],
      },
    },

    supportFile: 'cypress/support/component.ts',
  },
})
```

There is also a typo in the config shown in the question:

```ts
supportFile: 'cypress/cypress/support/component.ts'
```

should presumably be:

```ts
supportFile: 'cypress/support/component.ts'
```

You can also omit `indexHtmlFile` if you are using `cypress/support/component-index.html`, since that is Cypress's default location.

One current-version note: Cypress now supports Svelte 5 Component Testing with Vite, although Svelte support is still marked **alpha**.

Please mark this answer as accepted if it helped, thank you.
